test/resolve.test.ts

```diff
@@ -6,12 +6,12 @@ describe('resolveURL', () => {
     { input: [], out: '' },
     { input: ['/'], out: '/' },
     { input: ['/a'], out: '/a' },
-    { input: ['a', 'b'], out: 'a/b' },
-    { input: ['a', 'b/', 'c'], out: 'a/b/c' },
-    { input: ['a', 'b/', '/c'], out: 'a/b/c' },
-    { input: ['/a?foo=bar#123', 'b/', 'c/'], out: '/a/b/c/?foo=bar#123' },
+    { input: ['a', 'b'], out: 'b' },
```

Comment:
This is a breaking change and i think wrong. A path without leading slash (`b` vs `/b`) is not absolute that we are omitting previous part

Reply:
Yes, it is a breaking change. It depends on what this utility is doing. I think `resolveURL` _should_ behave this way, because of the word 'resolve'. It is also the RFC spec. And if we don't behave this way for leading slashes, we have no way to differentiate resolution for absolute vs relative URLs.

```html
<!-- page is https://test.com/a/b -->
<a href="c"> Link to https://test.com/a/c </a>
<a href="/c"> Link to https://test.com/c </a>
```

If we need the previous behaviour then I would suggest we rename the utility, as it wasn't _resolving_ URLs so much as combining segments and merging queries/hashes. (Out of curiosity, what was the use-case for this behaviour?)

Reply:
- If first arg (base or page) is absolute URL starting with protocol (`https://test.com/a/b`), it makes sense to omit `/a/b` but in case of joining `a` and `b` i think this is an odd behavior
- Historically UFO was based on URL to create utils which was wrong. We can keep closest URL behavior with `$URL` but rest of utils do not necessarily need this

We can discuss later what to do with `$URL` and `resolveURL`. But for me goal of this PR was to simply fix case of relative url handling and release as a hotfix :)

Reply:
I've now reverted breaking changes to `resolveURL` in favour of new `$URL.resolve()` utility that implements the RFC standard. And `resolveURL` and `joinURL` now support resolving relative segments :-D